pkg/controller/version/docker.go

```diff
@@ -24,62 +27,85 @@ func NewDockerVersionChecker(currentImage, currentImageId string, dockerConfig *
 }
 
 func (dockerVersionChecker *DockerVersionChecker) IsLatest() (bool, error) {
-	regex := regexp.MustCompile("(^docker-pullable:\\/\\/|\\:.*$|\\@sha256.*$)")
-	latestImageName := regex.ReplaceAllString(dockerVersionChecker.currentImage, "") + ":latest"
+	transportImageName := fmt.Sprintf("%s%s",
+		"docker://",
+		strings.TrimPrefix(
+			dockerVersionChecker.currentImageId,
+			"docker-pullable://"))
 
-	//Using ImageID instead of Image because ImageID contains digest of image that is used while Image only contains tag
-	reference, err := name.ParseReference(strings.TrimPrefix(dockerVersionChecker.currentImageId, "docker-pullable://"))
+	latestReference, err := alltransports.ParseImageName("docker://" + dockerVersionChecker.currentImage)
 	if err != nil {
 		return false, err
 	}
 
-	latestReference, err := name.ParseReference(latestImageName)
+	latestDigest, err := dockerVersionChecker.getDigest(latestReference)
 	if err != nil {
 		return false, err
 	}
 
-	registryURL := "https://" + reference.Context().RegistryStr()
-	authOption := getAuthOption(dockerVersionChecker.dockerConfig, registryURL)
-
-	latestDigest, err := getDigest(latestReference, authOption)
+	//Using ImageID instead of Image because ImageID contains digest of image that is used while Image only contains tag
+	currentReference, err := alltransports.ParseImageName(transportImageName)
+	//reference, err := name.ParseReference(strings.TrimPrefix(dockerVersionChecker.currentImageId, "docker-pullable://"))
```

Comment:
Shouldn't this line be used?

Reply:
Switched to `alltransports.ParseImageName(...)` because Skopeo uses the same call for it's `inspect` command